packages/components/src/internal/components/chart/ChartFieldOption.tsx

```diff
@@ -0,0 +1,230 @@
+
+import { ChartFieldInfo, ChartTypeInfo } from './ChartBuilderModal';
+import { getFieldDataType } from './utils';
+
+export const getSelectOptions = (
+    model: QueryModel,
+    chartType: ChartTypeInfo,
+    field: ChartFieldInfo
+): SelectInputOption[] => {
+    const allowableTypes = LABKEY_VIS.GenericChartHelper.getAllowableTypes(field);
+
+    return model.queryInfo
+        .getDisplayColumns(model.viewName)
+        .filter(col => {
+            const colType = getFieldDataType(col);
+            const hasMatchingType = allowableTypes.indexOf(colType) > -1;
+            const isMeasureDimensionMatch = LABKEY_VIS.GenericChartHelper.isMeasureDimensionMatch(
+                chartType.name,
+                field,
+                col.measure,
+                col.dimension
+            );
+            return hasMatchingType || isMeasureDimensionMatch;
+        })
+        .sort(naturalSortByProperty('caption'))
+        .map(col => ({ label: col.caption, value: col.fieldKey, data: col }));
+};
+
+export const shouldShowFieldOptions = (field: ChartFieldInfo, selectedType: ChartTypeInfo): boolean => {
+    const showForX = field.name === 'x' && (selectedType.name === 'scatter_plot' || selectedType.name === 'line_plot');
+    const showForY =
+        field.name === 'y' &&
+        (selectedType.name === 'scatter_plot' || selectedType.name === 'line_plot' || selectedType.name === 'box_plot');
+    return showForX || showForY;
+};
+
+const SCALE_TRANS_TYPES = [
+    { value: 'linear', label: 'Linear' },
+    { value: 'log', label: 'Log' },
+];
+
+const SCALE_RANGE_TYPES = [
+    { value: 'automatic', label: 'Automatic' },
+    { value: 'manual', label: 'Manual' },
+];
+
+const DEFAULT_SCALE_VALUES = { type: 'automatic', trans: 'linear' };
+
+interface ChartFieldOptionProps {
+    field: ChartFieldInfo;
+    fieldValue?: SelectInputOption;
+    model: QueryModel;
+    onScaleChange: (field: string, key: string, value: string | number, reset?: boolean) => void;
+    onSelectFieldChange: (name: string, value: any, selectedOption: SelectInputOption) => void;
+    scaleValues?: Record<string, string | number>;
+    selectedType: ChartTypeInfo;
+}
+
+export const ChartFieldOption: FC<ChartFieldOptionProps> = memo(props => {
+    const { field, model, selectedType, onSelectFieldChange, scaleValues, fieldValue, onScaleChange } = props;
+    const options = useMemo(() => getSelectOptions(model, selectedType, field), [model, selectedType, field]);
+    const isNumericType = useMemo(
+        () => LABKEY_VIS.GenericChartHelper.isNumericType(getFieldDataType(fieldValue?.data)),
+        [fieldValue?.data]
+    );
+    const showFieldOptions = isNumericType && shouldShowFieldOptions(field, selectedType);
+    const [scale, setScale] = useState<Record<string, string | number>>(scaleValues ?? {});
+    const invalidRange = useMemo(
+        () =>
+            scale.min !== undefined &&
```

Comment:
is it possible for us to restrict the values to either be defined or undefined so we can simplify this check and skip a null check?

Reply:
because of the different ways a chart config can be saved (i.e. App chart builder and LKS chart wizard) these empty values could end up as undefined or null, so leaving this as-is.